BaseStation/BaseStation/StartBaseStation.cs

```diff
@@ -20,17 +21,35 @@ public static class StartBaseStation
 		[STAThread]
         public static void Main(String[] args)
         {
-            //temporary example code
-            Server.BaseServer.Start(GamepadFactory.GetDriveGamePad());
-            Thread eventloop = new Thread(Server.BaseServer.EventLoop);
-            eventloop.IsBackground = true;
-            eventloop.Start();
-            Thread ipCameraControl = new Thread(() => CameraControl.Start(GamepadFactory.GetDriveGamePad()));
-            ipCameraControl.IsBackground = true;
-            ipCameraControl.Start();
+            BaseServer.Setup();
+            CameraControl.Setup();
+            new Thread(StartUpdates).Start();
             Application app = new Application();
-            app.Exit += (sd, ev) => Server.BaseServer.Shutdown();
+            app.Exit += (sd, ev) => StopUpdates();
 			app.Run(new MainWindow());
         }
+
+        private static void Update()
+        {
+            BaseServer.Update(Controller);
+            CameraControl.Update(Controller);
+            Thread.Sleep(1);
```

Comment:
May consider sleeping for a longer period of time than 1 ms just so we don't consume too many resources.

Reply:
https://stackoverflow.com/questions/6254703/thread-sleep-for-less-than-1-millisecond

According to this, calling `Thread.Sleep(1);` will typically sleep for 12-15ms anyway because Windows is not a real-time OS. (If it's on linux with mono, apparently it will sleep for exactly 1ms based on the comments in that thread, but in this case it's the BaseStation)

Reply:
I find that doing an update loop with a 1 ms sleep is plenty to keep CPU usage low (although I haven't tested this), since the actual work time for the Update methods won't even be close to a millisecond, meaning most of the time this loop will be sleeping.